test/functional/p2p_add_connections.py

```diff
@@ -0,0 +1,92 @@
+        assert_equal(info["connections_out"], num_out)
+
+    def run_test(self):
+        self.disconnect_nodes(0, 1)
+
+        self.log.info("Add 8 outbounds to node 0")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="outbound")
+
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+
+        self.log.info("Add 2 block-relay-only connections to node 1")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[1].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+
+        self.log.info("Add some inbound connections to node 1")
+        for i in range(5):
+            self.log.info("inbound: {}".format(i))
+            self.nodes[1].add_p2p_connection(P2PInterface())
+
+        self.log.info("Add 8 outbounds to node 1")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[1].add_outbound_p2p_connection(P2PInterface())
+
+        self.log.info("Check the connections opened as expected")
+        self.check_node_connections(node=0, num_in=0, num_out=10)
+        self.check_node_connections(node=1, num_in=5, num_out=10)
+
+        self.log.info("Disconnect p2p connections & try to re-open")
+        self.nodes[0].disconnect_p2ps()
+        self.check_node_connections(node=0, num_in=0, num_out=0)
+
+        self.log.info("Add 8 outbounds to node 0")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+        self.check_node_connections(node=0, num_in=0, num_out=8)
+
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+        self.check_node_connections(node=0, num_in=0, num_out=10)
+
+        self.log.info("Restart node 0 and try to reconnect to p2ps")
+        self.restart_node(0)
+
+        self.log.info("Add 8 outbounds to node 0")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface())
+        self.check_node_connections(node=0, num_in=0, num_out=8)
+
+        self.log.info("Add 2 block-relay-only connections to node 0")
+        for i in range(2):
+            self.log.info("block-relay-only: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="blockrelay")
+        self.check_node_connections(node=0, num_in=0, num_out=10)
```

Comment:
Since this is a sanity check, it would be nice if the values weren't the same as when we called this before restarting the node.  Maybe add only 7 outbounds and a few inbounds?

Reply:
kewl, done.